pkg/datapath/loader/netlink.go

```diff
@@ -692,3 +694,33 @@ func renameDevice(from, to string) error {
 
 	return nil
 }
+
+// DeviceHasTCProgramLoaded checks whether a give device tc filter/qdisc progs are missing.
+func DeviceHasTCProgramLoaded(ifc string) (bool, error) {
+	l, err := netlink.LinkByName(ifc)
+	if err != nil {
+		return false, fmt.Errorf("unable to find endpoint link by name: %w", err)
+	}
+
+	dd, err := netlink.QdiscList(l)
+	if err != nil {
+		return false, fmt.Errorf("unable to fetch qdisc list for endpoint: %w", err)
+	}
+	var found bool
+	for _, d := range dd {
+		if d.Type() == qdiscClsact {
+			found = true
+			break
+		}
+	}
+	if !found {
+		return false, nil
+	}
+
+	ff, err := netlink.FilterList(l, netlink.HANDLE_MIN_INGRESS)
+	if err != nil {
+		return false, fmt.Errorf("unable to fetch ingress filter list: %w", err)
+	}
+
+	return len(ff) > 0, nil
```

Comment:
Good point! Not sure what names we should check for as it may require additional context. 
It seems we are prefixing our prog names with `cil_*` is that what you were thinking? 
Also this feature will change when we leverage links to provide proper ownership for our bpf progs.